.github/workflows/haskell-ci.yml

```diff
@@ -0,0 +1,111 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  cabal:
+    name: cabal ${{ matrix.os }} / ghc ${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          # - macOS-latest
+          # - windows-latest
+        cabal: [3.4]
+        ghc:
+          - 8.6.5
+          # - 8.8.4
+          # - 8.10.7
+        # exclude:
+        #   # Test only with latest GHC on windows and macOS
+        #   - os: macOS-latest
+        #     ghc: 8.6.5
+        #   - os: macOS-latest
+        #     ghc: 8.8.4
+        #   - os: windows-latest
+        #     ghc: 8.6.5
+        #   - os: windows-latest
+        #     ghc: 8.8.4
+    services:
+      postgres:
+        image: postgres:12
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: helloci
+          POSTGRES_DB: pg_named
+        ports:
+          - 5432:5432
+    steps:
+      - uses: actions/checkout@v2
+        if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
+
+      - uses: haskell/actions/setup@v1
+        id: setup-haskell-cabal
+        name: Setup Haskell Cabal
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-version: ${{ matrix.cabal }}
+
+      - uses: actions/cache@v2
+        name: Cache ~/.cabal/store
+        with:
+          path: ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+          key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+
+      - name: Update
+        run: cabal update
```

Comment:
are you sure that these 2 won't conflict.

Reply:
These 2 what?

Reply:
2 steps. one will pull cache based on freeze but other will call update. Can it happen that update will update freez invalidating the cache?

Reply:
Anyway cabal update is not needed, because it's made as part of "setup cabal" step as I just learned.
Removed.

Reply:
good